### PR TITLE
Keep auth header untouched

### PR DESCRIPTION
**Pull request by dirkkul**
closes https://github.com/semi-technologies/weaviate-python-client/issues/177

### CODE REVIEW

File: weaviate/connect/connection.py [hunk 1]
```diff
@@ -6,14 +6,20 @@
 import datetime
 import os
 import time
+
+# request library uses simplejson (and its JSONDecodeError) if it is installed on the system and stdlib json otherwise.
+# See https://github.com/psf/requests/issues/4842 for more infos
+try:
+    from simplejson import JSONDecodeError
+except ImportError:
+    from json import JSONDecodeError
```
Comment (StefanBogdan) [inline weaviate/connect/connection.py:15]: Why not importing `requests.JSONDecodeError` without try except?
https://requests.readthedocs.io/en/latest/api/#requests.JSONDecodeError

File: weaviate/connect/connection.py [hunk 2]
```diff
@@ -110,15 +113,25 @@ def _create_session(self, auth_client_secret: Optional[AuthCredentials]) -> None
...
-            if auth_client_secret is not None:
+            # Some setups are behind proxies that return some default page - for example a login - for all requests.
+            # If the response is not json, we assume that this is the case and try unauthenticated access. Any auth
+            # header provided by the user is unaffected.
+            try:
                 resp = response.json()
+            except JSONDecodeError:
+                _Warnings.auth_cannot_parse_oidc_config(oidc_url)
+                self._session = requests.Session()
+                return
```
Comment (StefanBogdan) [inline weaviate/connect/connection.py:132]: Why do we assume that is the response is not decodable it works with authorization header passed by user?

File: weaviate/connect/connection.py (hunk 1 above)
Comment (dirkkul) [reply weaviate/connect/connection.py:15]: because nobody told me!

File: weaviate/connect/connection.py (hunk 2 above)
Comment (dirkkul) [reply weaviate/connect/connection.py:127]: We assume that there is no weaviate authentication if we cannot parse it, doesn't matter if users provide any header or not. Is there anything better to do?

Comment (StefanBogdan) [reply weaviate/connect/connection.py:127]: If there is no Weaviate authentication Weaviate returns 404 status code for `/.well-known/openid-configuration`. At least it was like this.

Comment (dirkkul) [reply weaviate/connect/connection.py:127]: Check the linked issue. :)

The user has a proxy that returns a response for all URLs that do not exists, eg every request returns status code 200

Comment (StefanBogdan) [reply weaviate/connect/connection.py:127]: Sorry, missed the Issue there. Makes sense. Then it can be merged.

Comment (dirkkul) [reply weaviate/connect/connection.py:127]: thanks!

**Pull request merged**